internal/app/machined/main.go

```diff
@@ -195,7 +194,13 @@ func run() error {
 	go runDebugServer(ctx)
 
 	// Schedule service shutdown on any return.
-	defer system.Services(c.Runtime()).Shutdown(ctx)
+	s := system.Services(c.Runtime())
+	defer func() {
+		shutdownCtx, shutdownCtxCancel := context.WithTimeout(context.Background(), time.Minute)
```

Comment:
This is required, because when we cancel the context, `Shutdown(context)` sequence below will never run properly.